### ISSUE TITLE
Sort files into format version specfic folders instead of Future/Current/Past

### ISSUE THREAD

**Issue by hf-kklein** [Hochfrequenz/edi_energy_scraper#106]
### Current Behaviour
The current default and only supported directory structure is:

- future contains all files valid in the future (relative to the time the script is run at)
- current contains all files valid as of now (the time the script is run)
- past contains all archived documents

This is ok, because it 1:1 reflects the structure found on edi-energy.de.

The directory structure as of now is:
```
- mirror_root
            |--future
                  |--- all document > now()
                  |    ...
            |--current
                  |--- all documents valid now
                  |    ...
            |--past
                  |--- all archived documents
                  |    ...
```

### Reason for a Change
But for archiving the documents and running further analysis scraping on them, it'd be good if we had a stable absolute instead of "relative", moving sorting.

So instead of FUTURE/CURRENT/PAST (actual behaviour right now) we prefer FV2404 (`FUTURE` as of January 2024), FV2310 (`CURRENT` as of January 2024) or FV2304, FV2210, FV2204, FV2110 ... (all `PAST` as of January 2024).

This will make things easier e.g. in https://github.com/Hochfrequenz/edi_energy_mirror/issues/138 or https://github.com/Hochfrequenz/kohlrahbi/issues/216.

### Requirement / Expected new Behaviour:
- [ ] edi_energy_scraper shall move files into folders named after [`EdifactFormatVersion`s](https://github.com/Hochfrequenz/mig_ahb_utility_stack/blob/5950d0b2c862e762ef7961dc5c951c99dca477c7/src/maus/edifact.py#L67) instead of the [`Epoch`s](https://github.com/Hochfrequenz/edi_energy_scraper/blob/e20f17474fc15f6719438f899a99dec5dfc5a355/src/edi_energy_scraper/epoch.py#L7) depending on the validity time range
- [ ] edi_energy_scraper shall also detect the [`EdifactFormat`](https://github.com/Hochfrequenz/mig_ahb_utility_stack/blob/5950d0b2c862e762ef7961dc5c951c99dca477c7/src/maus/edifact.py#L16) to which each file belongs
- [ ] edi_energy_scraper shall put general file which do not belong to a specific format into the respective format version folder

The new prefered directory structure shall be:
```
- mirror_root
            |--<...newer format versions...>
            |--FV2410
                  |--- all document valid for 2024-10-01 <= t < 2025-04-01
                  |---APERAK
                        |--- all respective APERAK files (MIG, AHB, ...)
                  |---COMDIS
                        |--- all respective COMDIS files (MIG, AHB, ...)
                  |--- other formats
                  |     ...
            |--FV2404
                  |--- all document valid for 2024-04-01 <= t < 2024-10-01
                  |---APERAK
                        |--- all respective APERAK files (MIG, AHB, ...)
                  |---COMDIS
                        |--- all respective COMDIS files (MIG, AHB, ...)
                  |--- other formats
                  |     ...
            |--FV2310
                  |--- all document valid for 2023-10-01 <= t < 2024-04-01
                  |---APERAK
                        |--- all respective APERAK files (MIG, AHB, ...)
                  |---COMDIS
                        |--- all respective COMDIS files (MIG, AHB, ...)
                  |--- other formats
                  |     ...
            |--FV2304
                  |--- all document valid for 2023-01-01 <= t < 2023-10-01
                  |---APERAK
                        |--- all respective APERAK files (MIG, AHB, ...)
                  |---COMDIS
                        |--- all respective COMDIS files (MIG, AHB, ...)
                  |--- other formats
                  |     ...
            |--<...older format versions...>
```

### How To / Detailed Instructions
- [x] Add [`maus`](https://pypi.org/project/maus/)>=0.4.1 as a dependency; Open a PR just for this https://github.com/Hochfrequenz/edi_energy_scraper/pull/110
- [x] Write a function that for a given file (`Path`) name determines a `tuple[EdifactFormatVersion, EdifactFormat|None]`. Therefore import EdifactFormatVersion and EdifactFormat from maus and use the utility-function [`get_edifact_format_version(datetime)`](https://github.com/Hochfrequenz/mig_ahb_utility_stack/blob/5950d0b2c862e762ef7961dc5c951c99dca477c7/src/maus/edifact.py#L87). Write a test for your function that is parametrized with all the filenames we currently encounter (see https://github.com/Hochfrequenz/kohlrahbi/issues/216#issuecomment-1902238107 or [manually create a list](https://github.com/Hochfrequenz/edi_energy_mirror/tree/master/edi_energy_de)). Open a PR just for this
- [ ] Now that everything is prepared, change the logic of edi_energy_scraper: Instead of saving the files to a directory that contains the epoch, save the files to a path as defined in the requirements above. Start with debugging the [existing logic](https://github.com/Hochfrequenz/edi_energy_scraper/blob/e20f17474fc15f6719438f899a99dec5dfc5a355/src/edi_energy_scraper/__init__.py#L96) (there are tests :)). Modify the existing unittests such that they match the adapted business logic. You can diecide if you want the new logic to be configurable via a flag or if it should just replace the old logic (the latter is probably easier). Open a PR for that
- [ ] Wait for #104 to be fixed or fix it yourself in a separate PR
- [ ] Once the last PR is merged, release a new minor version because this is a breaking change
- [ ] Use this new release and bump the edi_energy_scraper version used in [edi_energy_mirror](https://github.com/Hochfrequenz/edi_energy_mirror/blob/4fa1ed036c8cdec785e1437d7e2cfb243c5307cb/dependencies/mirror-requirements.in#L1)
- [ ] Also modify the kohlrahbi CI script there; This should fix https://github.com/Hochfrequenz/edi_energy_mirror/issues/138
- [ ] The last 2 change probably need 1 common PR

---

@hf-krechan anything else?

**Comment by hf-krechan** [comment]
I created a PR #107 to fix the release workflow.

**Comment by hf-krechan** [comment]
The PR #150 sorts the downloaded files into format version directories.
So this issue is done
